iidm/iidm-serde/src/test/java/com/powsybl/iidm/serde/extensions/SerializationNamesTest.java

```diff
@@ -0,0 +1,172 @@
+import static org.mockito.Mockito.*;
+
+/**
+ * @author Olivier Perrin {@literal <olivier.perrin at rte-france.com>}
+ */
+class SerializationNamesTest extends AbstractIidmSerDeTest {
+
+    public static final String LOAD_MOCK = "loadMock";
+    public static final String LOAD_ELEMENT_MOCK = "loadElementMock";
+
+    @Test
+    void unversionedExtensionTest() {
+        // Use the default extension version => also the default serialization name
+        AbstractExtensionSerDe<?, ?> serde = new NetworkSourceExtensionSerDe();
+        assertEquals(serde.getExtensionName(), serde.getSerializationName(serde.getVersion()));
+        Set<String> serializationNames = serde.getAlternativeSerializationNames();
+        assertTrue(serializationNames.isEmpty());
+    }
+
+    @Test
+    void getSerializationNameTest() {
+        // Use an extension version which serialization name is not the default
+        LoadMockSerDe serde = new LoadMockSerDe();
+        assertEquals(LOAD_MOCK, serde.getExtensionName());
+        assertEquals(LOAD_MOCK, serde.getSerializationName(serde.getVersion()));
+        assertEquals(LOAD_ELEMENT_MOCK, serde.getSerializationName("0.1"));
+        Set<String> serializationNames = serde.getAlternativeSerializationNames();
+        assertEquals(2, serializationNames.size());
+        assertTrue(serializationNames.containsAll(Set.of(LOAD_ELEMENT_MOCK, "loadEltMock")));
+    }
+
+    @Test
+    void realExtensionNameRoundTripTest() throws IOException {
+        // Use the default extension version => also the default serialization name
+        allFormatsRoundTripTest(getNetwork(), "/extensionName_1_2.xml", new ExportOptions().setVersion(IidmVersion.V_1_1.toString(".")));
+    }
+
+    @Test
+    void oldNameRoundTripTest() throws IOException {
+        // Use an extension version which serialization name is not the default
+        ExportOptions exportOptions = new ExportOptions()
+                .setVersion(IidmVersion.V_1_0.toString("."))
+                .addExtensionVersion("loadMock", "0.1");
+        allFormatsRoundTripTest(getNetwork(), "/extensionName_0_1.xml", exportOptions);
+    }
+
+    @Test
+    void importXiidmWithAnotherPrefixTest() throws URISyntaxException {
+        // Read a network with an old extension serialization name + a non-standard namespace prefix.
+        Network network = NetworkSerDe.validateAndRead(Paths.get(
+                Objects.requireNonNull(getClass().getResource("/extensionName_0_1_otherPrefix.xml")).toURI()));
+        assertNotNull(network.getLoad("Load1").getExtension(LoadMockExt.class));
+    }
+
+    @Test
+    void indicateExtensionAtImportTest() throws URISyntaxException {
+        // Read a network with an old extension serialization name
+        // To specify the extension to import, both the real extension name or the serialization name can be used.
+        Path file = Paths.get(Objects.requireNonNull(getClass().getResource("/extensionName_0_1_otherPrefix.xml")).toURI());
```

Comment:
It is not robust to convert a resource to a Path. You should use `getResourceAsInputStream` and load the network from it.